.github/workflows/main.yml

```diff
@@ -99,3 +99,33 @@ jobs:
         bundler-cache: true
     - name: Run Ruby tests with valgrind
       run: bundle exec rake test:valgrind
+
+  gcc-analyzer:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: head
+        bundler-cache: true
+    - name: Run build with gcc-analyzer enabled
+      run: |
+        sed -e  's/CFLAGS :=$/CFLAGS := -fanalyzer/g' -i Makefile
```

Comment:
This is fragile, but AFAIU currently it's not possible to pass CFLAGS to the `make` from the outside. Can create another target though.